web-wallet/src/lib/containers/MigrateContract/MigrateContract.svelte

```diff
@@ -171,29 +164,24 @@
       screenWidth = entry.contentRect.width;
     });
 
-    (async () => {
-      if (isConnected) {
-        await walletDisconnect();
+    const handleModalEvents = async (
+      /** @type {{ data: { event: string; }; }} */ e
+    ) => {
+      if (e.data.event === "CONNECT_SUCCESS") {
+        switchToSelectedChain();
+        connectedWalletBalance = await getBalance();
       }
+    };
 
-      modal.subscribeEvents(async (e) => {
-        if (e.data.event === "CONNECT_SUCCESS") {
-          switchToSelectedChain();
-          connectedWalletBalance = await getBalance();
-        }
-      });
+    resizeObserver.observe(document.body);
+    modal.subscribeEvents(handleModalEvents);
 
-      amountInput = document.querySelector(".migrate__input-field");
-
-      resizeObserver.observe(document.body);
-    })();
-
-    return () => resizeObserver.disconnect();
+    return () => {
+      resizeObserver.disconnect();
+    };
   });
 
-  onDestroy(async () => {
-    await walletDisconnect();
-  });
+  onDestroy(() => walletDisconnect());
```

Comment:
Please double check this change, because I remember that I suggested it as well in the past, but @deuch13 said that the disconnect needs to be awaited.

Reply:
It didn't make sense anyway, as the `walletDisconnect` is a method we have defined, that was just calling `disconnect(wagmiConfig)`.

I did check the docs for wagmi, indeed there they await the disconnect call, yet without any information on why.

Anyways, I made a change to our walletDisconnect method that 1) awaits the disconnect call, as per docs, 2) also calls disconnect on the modal instance.

ref: https://github.com/dusk-network/rusk/pull/3137/files#diff-527db978c77ff5858dc25843d437ffc58272abd046869a854e881c8fc3020340R68-R75